filter_plugins/oo_filters.py

```diff
@@ -128,14 +128,14 @@ def oo_select_keys(data, keys):
                 returns [1, 3]
         """
 
-        if not isinstance(data, dict):
-            raise errors.AnsibleFilterError("|failed expects to filter on a dict")
+        if not isinstance(data, dict) and not isinstance(data, object):
+            raise errors.AnsibleFilterError("|failed expects to filter on a dict or object")
```

Comment:
You are right, that hostvars is now an object. I updated the patch to check whether we are dealing with a Mapping.